e2e/tests/functional/recentObjects.e2e.spec.js

```diff
@@ -102,31 +101,117 @@ test.describe('Recent Objects', () => {
+
+        // Assert that the Clock parent folder has expanded and the Clock is visible)
+        await expect(folderTreeItem.locator('.c-disclosure-triangle')).toHaveClass(/--expanded/);
+        await expect(clockTreeItem).toBeVisible();
+
+        // Assert that the Clock treeitem is highlighted
+        await expect(clockTreeItem.locator('.c-tree__item')).toHaveClass(/is-targeted-item/);
+
+        // Wait for highlight animation to end
+        await waitForAnimations(clockTreeItem.locator('.c-tree__item'));
+
+        // Assert that the Clock treeitem is no longer highlighted
+        await expect(clockTreeItem.locator('.c-tree__item')).not.toHaveClass(/is-targeted-item/);
     });
-    test.fixme("Tests for context menu actions from recent objects", async ({ page }) => {
+    test("Persists on refresh", async ({ page }) => {
+        assertInitialRecentObjectsListState();
+        await page.reload();
+        assertInitialRecentObjectsListState();
     });
+    test("Displays objects and aliases uniquely", async ({ page }) => {
+        const mainTree = page.getByRole('tree', { name: 'Main Tree'});
+
+        // Navigate to the clock and reveal it in the tree
+        await page.goto(clock.url);
+        await page.getByTitle('Show selected item in tree').click();
+
+        // Right click the clock and create an alias using the "link" context menu action
+        const clockTreeItem = page.getByRole('tree', {
+            name: 'Main Tree'
+        }).getByRole('treeitem', {
+            name: clock.name
+        });
+        await clockTreeItem.click({
+            button: 'right'
+        });
+        await page.getByRole('menuitem', {
+            name: /Create Link/
+        }).click();
+        await page.getByRole('tree', { name: 'Create Modal Tree'}).getByRole('treeitem').first().click();
+        await page.getByRole('button', {
+            name: 'Save'
+        }).click();
+
+        // Click the newly created object alias in the tree
+        await mainTree.getByRole('treeitem', {
+            name: new RegExp(clock.name)
+        }).filter({
+            has: page.locator('.is-alias')
+        }).click();
+
+        // Assert that two recent objects are displayed and one of them is an alias
+        expect(await recentObjectsList.getByRole('listitem', { name: clock.name }).count()).toBe(2);
+        expect(await recentObjectsList.locator('.is-alias').count()).toBe(1);
+
+        // Assert that the alias and the original's breadcrumbs are different
+        const clockBreadcrumbs = recentObjectsList.getByRole('listitem', {name: clock.name}).getByRole('navigation');
+        expect(await clockBreadcrumbs.count()).toBe(2);
+        expect(await clockBreadcrumbs.nth(0).innerText()).not.toEqual(await clockBreadcrumbs.nth(1).innerText());
+    });
+    test("Enforces a limit of 20 recent objects", async ({ page }) => {
+        // Creating 21 objects takes a while, so increase the timeout
+        test.slow();
```

Comment:
this decorator used to work at one point so you could throw it onto the test itself. i.e.

```test.slow("Enforces a limit```

Reply:
[Not anymore, apparently.](https://playwright.dev/docs/api/class-test#test-slow-1)